### ISSUE TITLE
New visualisation of cross-slope properties in Antarctic margins?

### ISSUE THREAD

**Issue by taimoorsohail** [willaguiar/ASC_and_heat_transport#38]
I have written a piece of example code that plots ocean properties binned by isobath, as shown below. I have remapped the isobath bins (which are the same as st_ocean) by the cumulative area held by each bin, normalised. This produces a 'normalised distance from Antarctica', creating a more realistic slope representation. We can also multiply the normalised cumulative area by latitude to create a "pseudo-Latitude". Anyway, I was wondering what we all think about this? We can mask the ocean properties by regime before binning, which would give us a clear representation of the average T, S and rho in each regime, without having to resort to using individual sections.
 Let me know! 

![Avg_thermohaline_props](https://github.com/willaguiar/ASC_and_heat_transport/assets/20695740/410a4989-3f9e-4ccf-8ed7-d1c7718d0424)

**Comment by adele-morrison** [comment]
Love it! Would be great to apply this to u_along, temperature and density for temporal composites of months with strong or weak ASC.

**Comment by taimoorsohail** [comment]
Yep, it is easy enough to do with scalar tracer properties like temperature, and density, but u_along is harder. We would have to rotate the velocity vectors to be along- and across-slope for each isobath (the same isobaths as there are st_ocean values). I'm guessing this is an expensive code, considering we only have some selected isobaths for which u_along is available? The same applies for CSHT, by the way, with the added complexity of the zonal convergence term. I'm not sure how to handle u_along and CSHT, if there is any code that does this for multiple isobaths that would be great to see and incorporate.

**Comment by adele-morrison** [comment]
I think for ACC speed we could just use u in these zonally averaged transect plots. It won't give values that match u_along, but it should still show when / where the ACC is strong/weak.

And maybe we can use CSHT to define the temporal composites (i.e. all the times when CSHT is southward), and then just show a schematic arrow for CSHT, but not actual data.

**Comment by taimoorsohail** [comment]
Cool! I'll try this! By the way, the [xwmb](https://github.com/hdrake/xwmb/tree/main) package seems promising in quickly assessing transport across and along any given arbitrary section - might be faster than the methods we are currently using to calculate u_along and CSHT+ZC.

**Comment by adele-morrison** [comment]
Oh, good to know about xwmb! I'm looking forward to Henri's upcoming COSIMA talk about it.

**Comment by adele-morrison** [comment]
@taimoorsohail did you upload the code for this? I was going to make a new Figure 4 version using this method now, but am struggling to find the code.

**Comment by taimoorsohail** [comment]
Sorry Adele, I neglected to upload the relevant file before going on leave. I've uploaded it to my branch (taimoor) under the name "Circumpolar_average_example.ipynb". Let me know if you need any more help.
